copper-server-api/src/main/java/com/pogeyan/cmis/api/utils/Helpers.java

```diff
@@ -410,11 +414,25 @@ public static String splitFilterQuery(String filter) {
 		return result;
 	}
 
-	public static String[] getTypeMappedColumns(List<String> fieldAccess) {
-		String[] accessValues = fieldAccess.stream().map(t -> "propertyDefinition." + t).toArray(String[]::new);
-		String[] fieldAccessValues = Stream.concat(Arrays.stream(defaultProperty), Arrays.stream(accessValues))
-				.toArray(String[]::new);
-		return fieldAccessValues;
+	public static String[] getTypeMappedColumns(List<String> fieldAccess, IUserObject role, String typeId) {
+		if (role == null) {
+			return null;
+		}
+		if (!ArrayUtils.contains(basicTypes, typeId) && !isSystemUser(role)) {
```

Comment:
condense this if() condition with the prev if to return null, if its baseTypes or system user

Reply:
condensed the if condition